projects/storefrontapp-e2e-cypress/cypress/integration/b2b/order-approval/b2b-order-approval.e2e-spec.ts

```diff
@@ -0,0 +1,275 @@
+import { tabbingOrderConfig as config } from '../../../helpers/accessibility/b2b/tabbing-order.config';
+import * as orderApproval from '../../../helpers/b2b/b2b-order-approval';
+import * as sampleData from '../../../sample-data/b2b-order-approval';
+
+describe('B2B - Order Approval', () => {
+  before(() => {
+    cy.window().then((win) => win.sessionStorage.clear());
+  });
+
+  describe('Order Approval - Accessibility.', () => {
+    beforeEach(() => {
+      orderApproval.loginB2bApprover();
+    });
+
+    context('Approval List', () => {
+      it('should allow to navigate with tab key', () => {
+        orderApproval.approvalListTabbingOrder(config.orderApprovalList);
+      });
+    });
+
+    context('Approval Detail', () => {
+      it('should allow to navigate with tab key', () => {
+        orderApproval.approvalDetailTabbingOrder(config.orderApprovalDetail);
+      });
+    });
+
+    context('Approval Form', () => {
+      it('should allow to navigate with tab key', () => {
+        orderApproval.approvalFormTabbingOrder(config.orderApprovalForm);
+      });
+    });
+
+    context('Rejection Form', () => {
+      it('should allow to navigate with tab key', () => {
+        orderApproval.rejectionFormTabbingOrder(config.orderRejectionForm);
+      });
+    });
+  });
+  describe('B2B - Check order approval in Order details page for customer', () => {
+    it('should display order approval details in order details page', () => {
+      orderApproval.loginB2bUser();
+      orderApproval.getStubbedPendingOrderDetails();
+
+      cy.visit(`/my-account/order/${sampleData.ORDER_CODE}`);
+      assertPermissionResults(sampleData.pendingOrder);
+    });
+  });
+
+  describe('B2B - Order approval list and details for order approver', () => {
+    beforeEach(() => {
+      orderApproval.loginB2bApprover();
+    });
+
+    it('should display order approval list', () => {
+      orderApproval.getStubbedOrderApprovalList();
+      orderApproval.visitOrderApprovalListPage();
+      cy.wait('@order_approval_list').its('status').should('eq', 200);
+
+      cy.get('cx-order-approval-list a.cx-order-approval-value')
```

Comment:
Maybe there is a way to abstract these selectors into either a function or a block so that we don't repeat them?

Reply:
Done.